[TextField] Prevent runaway intervals and non-primary clicks (#1029)
Shopify/polaris-react: UNRELEASED.md

### Bug fixes

> - Fixed a bug with `type="number"` `TextField` where press-and-hold could increment or decrement infinitely ([#1029](https://github.com/Shopify/polaris-react/pull/1029))

Reviewer: ```suggestion
- Fixed `type="number"` `TextField` so press-and-hold doesn't increment or decrement infinitely ([#1029](https://github.com/Shopify/polaris-react/pull/1029))
```

https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md#how-to-write-a-changelog-entry

Author: ```suggestion
- Fixed `type="number"` `TextField` to prevent conditions where press-and-hold could increment or decrement infinitely ([#1029](https://github.com/Shopify/polaris-react/pull/1029))
```

Reviewer: Looks great!
